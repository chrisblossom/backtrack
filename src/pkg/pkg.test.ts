import path from 'path';

function getPkg() {
	const Backtrack = require('../initialize/initialize').Initialize;

	const backtrack = new Backtrack();

	return backtrack.pkg;
}

describe('pkg', () => {
	const cwd = process.cwd();
	let pkg: any;

	beforeEach(() => {
		const dir = path.resolve(__dirname, '__sandbox__/app1/');
		process.chdir(dir);
		jest.doMock('parent-module', () => () => path.join(dir, 'testing.js'));

		pkg = getPkg();
	});

	afterEach(() => {
		process.chdir(cwd);
	});

	test('handles missing sourceId', () => {
		let error;
		try {
			pkg.resolve(undefined, 'eslint');
		} catch (e) {
			error = e;
		} finally {
			expect(error).toMatchSnapshot();
		}
	});

	test('handles missing packageId', () => {
		let error;
		try {
			pkg.resolve('preset-01');
		} catch (e) {
			error = e;
		} finally {
			expect(error).toMatchSnapshot();
		}
	});

	test('handles unmatched resolve source with none available', () => {
		let error;
		try {
			expect.hasAssertions();

			pkg.resolveMap = {};

			pkg.resolve('preset-01', 'eslint');
		} catch (e) {
			error = e;
		} finally {
			expect(error).toMatchSnapshot();
		}
	});

	test('handles unmatched resolve source', () => {
		let error;
		try {
			pkg.resolve('preset-01', 'eslint');
		} catch (e) {
			error = e;
		} finally {
			expect(error).toMatchSnapshot();
		}
	});
});
