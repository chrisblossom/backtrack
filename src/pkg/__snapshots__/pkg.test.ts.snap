// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pkg handles missing packageId 1`] = `[Error: packageId is required]`;

exports[`pkg handles missing sourceId 1`] = `[Error: sourceId is required]`;

exports[`pkg handles unmatched resolve source 1`] = `[Error: sourceId "preset-01" is not valid. Available sourceIds: backtrack-preset-02, backtrack-preset-01]`;

exports[`pkg handles unmatched resolve source with none available 1`] = `[Error: sourceId "preset-01" is not valid. Available sourceIds: null]`;
